src/sentry/api/endpoints/auth_index.py

```diff
@@ -100,8 +103,17 @@ def put(self, request: Request):
             return Response(status=status.HTTP_401_UNAUTHORIZED)
 
         validator = AuthVerifyValidator(data=request.data)
+        has_auth_identity = False
         if not validator.is_valid():
-            return self.respond(validator.errors, status=status.HTTP_400_BAD_REQUEST)
+            auth_identity = AuthIdentity.objects.get(user_id=request.user.id)
+            if not auth_identity or not request.user.is_superuser:
+                return self.respond(validator.errors, status=status.HTTP_400_BAD_REQUEST)
+
+            has_auth_identity = True
+            # # TODO do we need to check if last sso verification was x amount of time?
```

Comment:
@JoshFerge @leedongwei For the users without passwords trying to enable active su, do we want to make sure they were last verified within a certain time (15 mins before) otherwise we ask them to re-authenticate or is the default sso session time (20 hours?) good enough?